lib/core/uri_parser.js

```diff
@@ -285,8 +285,8 @@ function applyConnectionStringOption(obj, key, value, options) {
     }
   }
 
-  if (key === 'readpreferencetags' && Array.isArray(value)) {
-    value = splitArrayOfMultipleReadPreferenceTags(value);
+  if (key === 'readpreferencetags') {
+    value = Array.isArray(value) ? splitArrayOfMultipleReadPreferenceTags(value) : [value];
```

Comment:
There's logic in lib/core/topologies/read_preference.js in the constructor to handle this, do we wanna make sure we consolidate somehow?

Reply:
I think it's a different concern, this is parsing the values from the connection string, that is logic for dealing with user-constructed `ReadPreference` instances. I think we need validation in both places.

Reply:
Yea that makes sense 👍